server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1641,13 +1654,28 @@ public BlobContainer shardContainer(IndexId indexId, int shardId) {
      *
      * @param repositorySettings repository settings
      * @param setting            setting to use to configure rate limiter
-     * @return rate limiter or null of no throttling is needed
+     * @param warnIfOverRecovery log a warning if rate limit setting is over the effective recovery rate limit
+     * @return rate limiter or null if no throttling is needed
      */
-    private static RateLimiter getRateLimiter(Settings repositorySettings, Setting<ByteSizeValue> setting) {
+    private RateLimiter getRateLimiter(Settings repositorySettings, Setting<ByteSizeValue> setting, boolean warnIfOverRecovery) {
         ByteSizeValue maxSnapshotBytesPerSec = setting.get(repositorySettings);
         if (maxSnapshotBytesPerSec.getBytes() <= 0) {
             return null;
         } else {
+            if (warnIfOverRecovery && recoverySettings.rateLimiter() != null) {
+                double effectiveRecoverySpeed = recoverySettings.rateLimiter().getMBPerSec();
+                if (maxSnapshotBytesPerSec.getMbFrac() > effectiveRecoverySpeed) {
+                    logger.warn(
+                        "[{}] repository snapshot rate limit [{}={}] will be capped by the effective recovery rate limit [{}] per sec",
+                        metadata.name(),
+                        setting.getKey(),
+                        maxSnapshotBytesPerSec,
+                        effectiveRecoverySpeed > 1.0
+                            ? Strings.format1Decimals(effectiveRecoverySpeed, "mb")
```

Comment:
We can log the byte size value read previously

Reply:
Indeed I did that now.